tardis/simulation/base.py

```diff
@@ -310,6 +311,24 @@ def legacy_update_spectrum(self, model, no_of_virtual_packets):
             model.spectrum_virtual.update_luminosity(
                 model.montecarlo_virtual_luminosity)
 
+    def legacy_set_final_model_properties(self, model):
+        """Sets additional model properties to be compatible with old model design
+
+        The runner object is given to the model and other packet diagnostics are set.
+
+        Parameters
+        ----------
+        model: ~tardis.model.Radial1DModel
+
+        Returns
+        -------
+            : None
+
+        """
+
+        #pass the runner to the model
+        model.runner = self.runner
+        #TODO: pass packet diagnostic arrays
```

Comment:
True - but to have full legacy suppport, the model should have some of these arrays (with the appropriate cuts) directly, see my comment below